crates/bevy_render/src/renderer/raw_vulkan_init.rs

```diff
@@ -0,0 +1,108 @@
+use bevy_ecs::resource::Resource;
+use bevy_platform::collections::HashSet;
+use std::{
+    any::{Any, TypeId},
+    sync::Arc,
+};
+use thiserror::Error;
+use wgpu::{
+    hal::api::Vulkan, Adapter, Device, DeviceDescriptor, Instance, InstanceDescriptor, Queue,
+};
+
+/// When the `raw_vulkan_init` feature is enabled, these settings will be used to configure the raw vulkan instance.
+#[derive(Resource, Default, Clone)]
+pub struct RawVulkanInitSettings {
+    pub create_instance_callbacks: Vec<
```

Comment:
These callbacks relate to the safety invariants in the unsafe `Vulkan::init_with_callbacks`, so registering a new callback should be unsafe.

Reply:
Fixed